### PR TITLE
Update installation docs

### PR DESCRIPTION
**Pull request by knelli2**
Small things I found while setting up a new environment

- Blaze version was out of date.
- If installing dependencies yourself, using the single-include header for catch is much easier.
- Singularity version of Python affects git hooks, but if you don't have the same version as the container it creates problems. ~Thus force python2 in the suggested build commands.~ So add a note explaining this so the user is aware.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: docs/Installation/Installation.md
Comment (wthrowe) [inline docs/Installation/Installation.md]: Some people won't have python 2 these days.  Distros are starting to drop it.

Comment (knelli2) [reply docs/Installation/Installation.md]: So then do we just leave it up to the user to figure out their python dependencies?

Comment (nilsdeppe) [reply docs/Installation/Installation.md]: Pretty much. Most people should be using python3 nowadays, but it's annoying that python decides that any minor version change means they can change anything and everything in the language. Python stability basically disappeared in 3.x

Comment (knelli2) [reply docs/Installation/Installation.md]: Ok instead of changing it in the example cmake command, I'll keep a note saying users should be aware of this.

Comment (nilsdeppe) [reply docs/Installation/Installation.md]: Yep, sounds reasonable to me :)

**Comment by knelli2** [comment]
Ok updated!

**Review by nilsdeppe — DISMISSED** [review]
I'm happy with this. Thanks for updating this info :)

File: docs/Installation/Installation.md
Comment (nilsvu) [inline docs/Installation/Installation.md]: - Would you mind updating the link to `https://github.com/catchorg/Catch2`?
- I'm not so sure we should add this recommendation. For instance, for me it was easiest to install catch through my package manager (macOS). On Minerva I installed it with Spack. I think we can't give a general recommendation here how to install things, but only on the installation options (like "disable mpi" for HDF5 and libsharp below). Which issue did you run into that prompted you to add this note? If you feel like adding a note to help others with that issue, perhaps phrase it more specific to that issue?
- Note that Catch is developing v3, which will no longer be header-only: https://github.com/catchorg/Catch2/blob/devel/docs/migrate-v2-to-v3.md#top

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: docs/Installation/Installation.md
Comment (knelli2) [reply docs/Installation/Installation.md]: - The current link redirects you to Catch2, but I'll still update it
- I was installing all the dependencies on my machine and inside catch there is `include/catch.cpp` and `single_include/catch2/catch.hpp`. I tried the first option but got a lot of errors when compiling about missing headers. The second option (single include) is what we are supposed to use and what is used inside the containers I believe.
- When (if) we switch to v3, then we can update docs. I don't think that should matter for this because it's v2.

**Comment by yoonso0-0** [comment]
I found doxygen link also broken and would be good to include in this PR (`https://www.doxygen.nl/index.html`).

File: docs/Installation/Installation.md
Comment (yoonso0-0) [reply docs/Installation/Installation.md]: Their `devel` branch is already in v3 and has a bit different source structure with v2.x. Should we add a phrase like '2.8.0 or later but not 3.x'?

Comment (knelli2) [reply docs/Installation/Installation.md]: As long as we don't yet support v3, then yeah I think that's ok.

**Comment by knelli2** [comment]
@nilsleiffischer reworded that help phrase and updated catch link
@yoonso0-0 updated doxygen link

Rebased on latest develop and squash changes directly.

**Review by nilsvu — DISMISSED** [review]

File: docs/Installation/Installation.md
Comment (nilsvu) [reply docs/Installation/Installation.md]: ok the way you formulated this looks fine to me. If you want to include @yoonso0-0's suggestion to exclude v3 then squash that in as well. No need to wait for my re-approval if someone gets to review this before me.

**Comment by knelli2** [comment]
Included @yoonso0-0 's suggestion of excluding catch v3

**Review by yoonso0-0 — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**
